src/content/blog/whats-new-june-2024.mdx

```diff
@@ -0,0 +1,151 @@
+- 📄 [Adding search to static Astro sites](https://www.thomasledoux.be/blog/search-static-astro-website) by @thomasledoux1
+- 📄 [Integrate Pagefind's Search with Astro: A Complete Setup Guide](https://syntackle.com/blog/pagefind-search-in-astro-site/) by @murtuzaalisurti
+- 🎙️ [title]() by @
+- 🎙️ [title]() by @
+- 📺 [title]() by @
+
+ ... and don't miss the latest tips from [Astro Tips](https://astro-tips.dev) such as [how to make a dynamic footer date for static sites](https://astro-tips.dev/recipes/dynamic-footer-date-for-static-websites/) and tutorials from [Lexingon Themes](https://lexingtonthemes.com/tutorials/) such as [How to create a swatch clock with Tailwind CSS]( https://lexingtonthemes.com/tutorials/how-to-create-a-swatch-clock-with-tailwind-css/)
+
+
+### Tips & Tools
+
+The latest community-built utilities and integrations to help you build with Astro.
+
+- [Astrolace](https://github.com/MatthiesenXYZ/Astrolace) - An easy way to use Shoelace.style within your Astro project by @matthiesenxyz
+- [Starlight View Modes](https://starlight-view-modes.trueberryless.org/) - Starlight plugin adding view mode capabilities to your documentation website by @trueberryless
+- [Starlight Utils](https://starlight-utils.pages.dev/) - Utilities to use with your 🌟 Starlight site by @lorenzolewis
+- [Content Utils Git Time](https://inox-tools.vercel.app/content-utils/git-time) - A library to retrieve last updated dates for your content from your Git history by @Fryuni
+- [Astro Command Palette](https://github.com/pauchiner/astro-command-palette) - A lightweight, extensible, and fast command palette for Astro by @pauchiner
+
+
+### Themes & Templates
+
+Several new Astro themes were added to the [Astro theme directory](https://astro.build/themes/) this month. Try them out!
+
+ <ThemeGrid themes={[ 'starfunnel', 'astro-big-doc', 'astro-kreatif-landing',]} />
+
+Also check out:
+
+- [Vendure Astro Storefront](https://github.com/a-atalla/vendure-astro-shop) by @a.atalla
+- [Astro Aceternity](https://github.com/calrloco/astro-aceternity) by @CarloB
+- [Eva's Astro Starter](https://github.com/evadecker/eva-astro-starter) by @evadecker
+- [Academic Project Starter Template](https://academic-project-astro-template.vercel.app/)
+
+### Websites
+
+Sites of the month:
+
+- [Astro ZeroJS View Transitions](https://astro-zerojs-transitions.vercel.app/), a Spotify clone using Astro's view transitions with 0JS.
+- Not just a website, [**an experience**](https://dracarys.robertborghesi.is/)! A WebGL, Three.js experiment on GPGPU simulations, particles, postprocessing, shaders, water, modelling, audio, lightning and dragons... your typical Astro site. 🔥🧊🐉 - 🔊 ON for full effect! (It's so wild, we cannot display a preview!)
+
+
+export const siteofmonth = [ 
+  { url: 'https://astro-zerojs-transitions.vercel.app/', title: '0JS view transitions'},
+  { url: 'https://dracarys.robertborghesi.is/', title: 'Dracarys - Robert Borghesi LAB'},
```

Comment:
Flagging that this one isn’t rendering properly — it’s too complex a site for 11ty’s screenshot service to capture
<img width="1042" alt="image" src="https://github.com/withastro/astro.build/assets/357379/2b0ff5ee-dbc7-4c3d-9f1c-5a682f1b8c37">

Reply:
Yes, I took a custom screenshot for the showcase, but since these "sites of the week" are still the old component, I hadn't figured out what to do here yet.

Reply:
Added a new `<ShowcaseGridById />` component for these use cases where we want to specify a number of specific sites. Used `id` (i.e. filename) as the keys here as these are easy to work out vs. `slug` which is processed by Astro and not really used anywhere.

I also added an `exclude` prop to the monthly grid component to allow us to then skip including sites featured in the sites of the month in the main grid.

Full usage looks something like:

```mdx
export const monthlySites = [
  'astro-zerojs-transitions.vercel.app.md',
  'dracarys.robertborghesi.is.md',
  'stickly.pages.dev.md',
];

<ShowcaseGridById ids={monthlySites} />

<MonthlyShowcaseGrid startDate="2024-06-01" endDate="2024-06-30" exclude={monthlySites} />
```

**N.B.** the zero JS transitions site wasn’t yet in the gallery, so I’ve commented it out and dropped the link in the GitHub Disussion so it’s picked up when we next run the showcase scraper.

Reply:
Just confirming that it's been picked up, and works fine now!